### ISSUE TITLE
Incorrect injection label behavior and incorrect docs

### ISSUE THREAD

**Issue by davtex** [istio/istio#39005]
### Bug Description

I have a similar problem as the guy here: [Issue 33618](https://github.com/istio/istio/issues/33618
)

I use ArgoCD to enable/disable Istio injection by setting "istio-injection" label to enabled/disabled. This works nicely as long as we want to inject all pods in given namespace. As soon as you want to enable injection for one pod only (e.g. for Prometheus in order to generate/export client certificates), you run into issues.

I cannot use istio-injection=enabled as that would inject all pods. I could theoretically set pod label to false for all the other pods in the namespace, but I consider that to be impractical, especially when you run 10+ other applications in the same NS, which is quite common for e.g. "monitoring" NS.

I cannot use istio-injection=disabled as that takes precedence over pod-level labels.

Now to the best of my knowledge, it's impossible to remove labels when applying YAML in Kubernetes -> we can only add or change with tools like ArgoCD.

So the obvious choice here is to set istio-injection=abcdefgh -> set to random value that is neither enabled or disabled. According to documentation this should work as according to [docs](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/), valid values for label are enabled/disabled. This is incorrect -> the way mutating webhook is written, any value except "enabled" is considered to be disabled, which breaks this otherwise-possible workaround.

TL;DR
It is not possible to remove injection label from NS with automation tools such as ArgoCD. At the same time, if we want to handle injection on pod-level instead of NS-level, the only way to make it work is if NS-level label is missing.

Only a small change in mutatingwebhook is required to rectify this -> it was described by the bloke in issue I referenced in the beginning.

### Version

```prose
istioctl version
client version: 1.12.2
control plane version: 1.13.2
data plane version: 1.13.2 (103 proxies), 1.10.1 (1 proxies)

kubectl version --short
Client Version: v1.22.5
Server Version: v1.21.5-eks-bc4871b
```

**Comment by litong01** [comment]
@davtex are you saying setting up a pod with `sidecar.istio.io/inject: "true"` not working? 

If this is indeed the case, we are so broken. here is what I tried

1. create a namespace named sample
2. deploy a service and a deployment as follows, notice the injection label for the pod:
```
cat << EOF | kubectl apply -n sample -f -
apiVersion: v1
kind: Service
metadata:
  name: helloworld
  labels:
    app: helloworld
    service: helloworld
spec:
  ports:
  - port: 5000
    name: http
  selector:
    app: helloworld
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: helloworld-v1
  labels:
    app: helloworld
    version: v1
spec:
  replicas: 1
  selector:
    matchLabels:
      app: helloworld
      version: v1
  template:
    metadata:
      labels:
        app: helloworld
        version: v1
        sidecar.istio.io/inject: "true"
    spec:
      containers:
      - name: helloworld
        image: docker.io/istio/examples-helloworld-v1
        resources:
          requests:
            cpu: "100m"
        env:
        - name: FLASK_RUN_PORT
          value: "5000"
        - name: FLASK_RUN_HOST
          value: "::"
        - name: FLASK_ENV
          value: "development"
        - name: FLASK_APP
          value: "/opt/microservices/app.py"
        command: ["flask", "run"]
        imagePullPolicy: IfNotPresent #Always
        ports:
        - containerPort: 5000
EOF
```

with the namespace `sample` not  having the injection label, the proxy pod get injected correctly. This is called pod level injection, which should have been working for many releases. Can you try the above?

Also I am not sure why it is impossible to remove a label from namespace using ArgoCD. Cannot ArgoCD run a command like this `kubectl label namespace sample istio-injection-`?
